File: src/main/java/inno/jago/ast/expression/UnaryExpressionNode.java
package inno.jago.ast.expression;

import lombok.Getter;
import lombok.Setter;

/*
unaryExpr           :   primaryExpr | unary_op unaryExpr;
 */

@Getter
@Setter
public class UnaryExpressionNode extends ExpressionNode {
    private String operator;
    private ExpressionNode operand;

    public UnaryExpressionNode(String operator, ExpressionNode operand) {
        if (operator == null ) {
            System.out.println("String operator should not be null");
            return;
        }
        if (operand == null) {
            System.out.println("ExpressionNode operand should not be null");
            return;
        }

        this.operator = operator;
        this.operand = operand;

        this.operand.setParent(this);
    }

}

File: src/main/kotlin/inno/jago/ast/expression/unary_expression/UnaryExpressionNode.kt
package inno.jago.ast.expression.unary_expression

import inno.jago.ast.expression.ExpressionNode
import inno.jago.lexer.Pos
import java.util.function.UnaryOperator

open class UnaryExpressionNode(pos: Pos,
                               val operatorNode: UnaryOperatorNode?,
                               val unaryExpressionNode: UnaryExpressionNode
): ExpressionNode(pos)
